viseron/components/gotify/__init__.py

```diff
@@ -0,0 +1,319 @@
+"""Gotify notifications for Viseron."""
+
+from __future__ import annotations
+
+import asyncio
+import base64
+import logging
+from threading import Thread
+from typing import TYPE_CHECKING
+
+import cv2
+import requests
+import voluptuous as vol
+
+from viseron.const import VISERON_SIGNAL_SHUTDOWN
+from viseron.domains.camera.const import EVENT_RECORDER_START
+from viseron.domains.camera.recorder import EventRecorderData
+from viseron.helpers import escape_string
+from viseron.helpers.logs import SensitiveInformationFilter
+from viseron.helpers.validators import CoerceNoneToDict
+
+from .const import (
+    COMPONENT,
+    CONFIG_CAMERAS,
+    CONFIG_DETECTION_LABEL,
+    CONFIG_DETECTION_LABEL_DEFAULT,
+    CONFIG_GOTIFY_PRIORITY,
+    CONFIG_GOTIFY_PRIORITY_DEFAULT,
+    CONFIG_GOTIFY_TOKEN,
+    CONFIG_GOTIFY_URL,
+    CONFIG_SEND_THUMBNAIL,
+    DESC_CAMERA_DETECTION_LABEL,
+    DESC_CAMERA_SEND_THUMBNAIL,
+    DESC_CAMERAS,
+    DESC_COMPONENT,
+    DESC_DETECTION_LABEL,
+    DESC_GOTIFY_PRIORITY,
+    DESC_GOTIFY_TOKEN,
+    DESC_GOTIFY_URL,
+    DESC_SEND_THUMBNAIL,
+)
+
+if TYPE_CHECKING:
+    from viseron import Event, Viseron
+
+LOGGER = logging.getLogger(__name__)
+
+CAMERA_SCHEMA = vol.Schema(
+    {
+        vol.Optional(
+            "detection_label",
+            description=DESC_CAMERA_DETECTION_LABEL,
+        ): str,
+        vol.Optional(
+            "send_thumbnail",
+            description=DESC_CAMERA_SEND_THUMBNAIL,
+        ): bool,
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+CONFIG_SCHEMA: vol.Schema = vol.Schema(
+    {
+        vol.Required(COMPONENT, description=DESC_COMPONENT): {
+            vol.Required(CONFIG_GOTIFY_URL, description=DESC_GOTIFY_URL): str,
+            vol.Required(CONFIG_GOTIFY_TOKEN, description=DESC_GOTIFY_TOKEN): str,
+            vol.Optional(
+                CONFIG_GOTIFY_PRIORITY,
+                description=DESC_GOTIFY_PRIORITY,
+                default=CONFIG_GOTIFY_PRIORITY_DEFAULT,
+            ): vol.All(vol.Coerce(int), vol.Range(min=1, max=10)),
+            vol.Optional(
+                CONFIG_DETECTION_LABEL,
+                description=DESC_DETECTION_LABEL,
+                default=CONFIG_DETECTION_LABEL_DEFAULT,
+            ): str,
+            vol.Optional(
+                CONFIG_SEND_THUMBNAIL, description=DESC_SEND_THUMBNAIL, default=False
+            ): bool,
+            vol.Required(CONFIG_CAMERAS, description=DESC_CAMERAS): {
+                vol.Any(str): vol.All(CoerceNoneToDict(), CAMERA_SCHEMA),
```

Comment:
Thanks, i think the issue you are having with the `config.json` stems from this line.
Since its a `vol.Any` declaration, the script thinks its supposed to be a select-type config option.

Try this instead and see if it works better:
```python
from viseron.helpers.validators import CameraIdentifier, CoerceNoneToDict
```

```python
vol.Required(CONFIG_CAMERAS, description=DESC_CAMERAS): {
    CameraIdentifier(): vol.All(CoerceNoneToDict(), CAMERA_SCHEMA)
},
```